integrate extern/{storage-fsm,sector-storage} into lotus source tree

raulk opened the pull request:
* Move `extern/storage-fsm` into `storage/sealing` package; drop replace directive.
* Move `extern/sector-storage` into `storage/sector` package; drop replace directive.
* Drop unneeded import aliases, because package names are consistent now.
* rm README, CircleCI config, LICENSES, etc. and other irrelevant artifacts from migrated repos.
* Introduce a `go generate` directive for CBOR generation in `storage/sealing`. Make `Makefile` call it.
* Fix lint issues in sector-storage.

vyzo commented:
test failure seems real and not a fluke; some file is missing:
```
--- FAIL: TestDownloadParams (0.00s)
panic: open ../parameters.json: no such file or directory [recovered]
	panic: open ../parameters.json: no such file or directory

goroutine 15 [running]:
testing.tRunner.func1.1(0x245ca40, 0xc000290630)
	/usr/local/go/src/testing/testing.go:988 +0x30d
testing.tRunner.func1(0xc0001b37a0)
	/usr/local/go/src/testing/testing.go:991 +0x3f9
panic(0x245ca40, 0xc000290630)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/filecoin-project/lotus/storage/sector/ffiwrapper.getGrothParamFileAndVerifyingKeys(0x800)
	/home/circleci/project/storage/sector/ffiwrapper/sealer_test.go:227 +0x113
github.com/filecoin-project/lotus/storage/sector/ffiwrapper.TestDownloadParams(0xc0001b37a0)
	/home/circleci/project/storage/sector/ffiwrapper/sealer_test.go:245 +0x71
```

raulk commented:
@vyzo hopefully this will fix it: https://github.com/filecoin-project/lotus/pull/3089/commits/4f6d01d2ea3f6804e787290d588b9e36b2970681

raulk commented:
@vyzo I suspect that if the test is downloading proof parameters, it has the potential to be a time sink. We may need to create a different CircleCI job for this.

vyzo commented:
yeah, by no means ideal.

raulk commented:
@vyzo all passing here; looks like the last failures were flaky tests.

raulk commented:
@magik6k here's a version that keeps `storage-fsm` (now renamed to `storage-sealing`) and `sector-storage` under `extern`. I also reverted the `sectorstorage=>sector` package rename.

With this, your review comment should be addressed.

magik6k commented on extern/sector-storage/fsutil/filesize_unix.go:
```diff
@@ -24,6 +24,6 @@ func FileSize(path string) (SizeInfo, error) {
 	//  See https://www.gnu.org/software/libc/manual/html_node/Attribute-Meanings.html
 	return SizeInfo{
-		int64(stat.Blocks) * 512, // NOTE: int64 cast is needed on osx
+		stat.Blocks * 512, // NOTE: int64 cast is needed on osx
```
This cast needs to be here

raulk replied:
Thanks for catching this! Will revert and add a `// nolint` here.

raulk commented:
@magik6k fixed.

fadeAce commented:
> * `sector` isn't a good name for a subsystem which is handling storage and is scheduling sealing operations
>   
>   * There are probably better names, but I'd keep `sectorstorage` for now, also minimizes the diff size
> * I'd keep those packages in `extern/`, the intention is to extract them back out of lotus when GFC development resumes

it's better to be called manager cause it's a scheduler and sector manangement executor

fadeAce commented:
> * Move `extern/storage-fsm` into `storage/sealing` package; drop replace directive.
> * Move `extern/sector-storage` into `storage/sector` package; drop replace directive.
> * Drop unneeded import aliases, because package names are consistent now.
> * rm README, CircleCI config, LICENSES, etc. and other irrelevant artifacts from migrated repos.
> * Introduce a `go generate` directive for CBOR generation in `storage/sealing`. Make `Makefile` call it.
> * Fix lint issues in sector-storage.

what I do suspect is that will this change to happen in master & mainnet release ? Moving a mod to subtree if it's needed to be done why not drop storage-fsm project in repo
